res/controllers/Hercules-DJControl-Inpulse-200-script.js

```diff
@@ -0,0 +1,182 @@
+// DJControl_Inpulse_200_script.js
+//
+// ***************************************************************************
+// * Mixxx mapping script file for the Hercules DJControl Inpulse 200.
+// * Author: DJ Phatso, contributions by Kerrick Staley
+// * Version 1.2 (March 2020)
+// * Forum: https://www.mixxx.org/forums/viewtopic.php?f=7&t=12592
+// * Wiki: https://mixxx.org/wiki/doku.php/hercules_djcontrol_inpulse_200
+//
+// Changes to v1.2
+// - Code cleanup.
+//
+// Changes to v1.1
+// - Fix seek-to-start and cue-master behavior.
+// - Tweak scratch, seek, and bend behavior.
+// - Controller knob/slider values are queried on startup, so MIXXX is synced.
+// - Fixed vinyl button behavior the first time it's pressed.
+//
+// v1.0 : Original forum release
+//
+// TO DO: Functions that could be implemented to the script:
+//
+//  FX:
+//  - See how to preselect effects for a rack
+//
+//*************************************************************************
+
+var DJCi200 = {};
+///////////////////////////////////////////////////////////////
+//                       USER OPTIONS                        //
+///////////////////////////////////////////////////////////////
+
+// How fast scratching is.
+DJCi200.scratchScale = 1.0;
+
+// How much faster seeking (shift+scratch) is than scratching.
+DJCi200.scratchShiftMultiplier = 4;
+
+// How fast bending is.
+DJCi200.bendScale = 1.0;
+
+// Other scratch related options
+DJCi200.kScratchActionNone = 0;
+DJCi200.kScratchActionScratch = 1;
+DJCi200.kScratchActionSeek = 2;
+DJCi200.kScratchActionBend = 3;
+
+DJCi200.init = function() {
+    // Scratch button state
+    DJCi200.scratchButtonState = true;
+    // Scratch Action
+    DJCi200.scratchAction = {
+        1: DJCi200.kScratchActionNone,
+        2: DJCi200.kScratchActionNone
+    };
+
+    //Turn On Vinyl buttons LED(one for each deck).
+    midi.sendShortMsg(0x91, 0x03, 0x7F);
+    midi.sendShortMsg(0x92, 0x03, 0x7F);
+
+    //Turn On Browser button LED
+    midi.sendShortMsg(0x90, 0x04, 0x05);
+
+    //Softtakeover for Pitch fader
+    engine.softTakeover("[Channel1]", "rate", true);
+    engine.softTakeover("[Channel2]", "rate", true);
+    engine.softTakeoverIgnoreNextValue("[Channel1]", "rate");
+    engine.softTakeoverIgnoreNextValue("[Channel2]", "rate");
+
+    //Set effects Levels - Dry/Wet
+    engine.setParameter("[EffectRack1_EffectUnit1_Effect1]", "meta", 0.6);
+    engine.setParameter("[EffectRack1_EffectUnit1_Effect2]", "meta", 0.6);
+    engine.setParameter("[EffectRack1_EffectUnit1_Effect3]", "meta", 0.6);
+    engine.setParameter("[EffectRack1_EffectUnit2_Effect1]", "meta", 0.6);
+    engine.setParameter("[EffectRack1_EffectUnit2_Effect2]", "meta", 0.6);
+    engine.setParameter("[EffectRack1_EffectUnit2_Effect3]", "meta", 0.6);
+    engine.setParameter("[EffectRack1_EffectUnit1]", "mix", 1);
+    engine.setParameter("[EffectRack1_EffectUnit2]", "mix", 1);
```

Comment:
This has nothing to do with the controller itself. afaik this should not be included in the official mapping. @Holzhaus opinions?

Reply:
Yes, true. I vaguely remember pointing out similar code in one of the past controller mapping PRs, but I can't find it anymore. I don't know how we proceeded there.

Reply:
I have the same issue in #2546 multiple times. I think the primary issue is that people abuse the controller script for some quick automation in JS. IMO the  best solution to this issue would be, that we create a second JS environment that is controller independent and let people do some basic scripting by manipulating COs that way. There are some usecases that are very userspecific and can be implemented solely via JS and the current scripting capabilities via CO's. Due to them being so user specific, it wouldn't make sense to implement them in C++ and merge them into Mixxx but many users are also not tech savvy enough to maintain their own fork. It would make sense for them to have a dynamic scripting environment.
Should I create a Zulip topic to discuss this further?

Reply:
> Should I create a Zulip topic to discuss this further?

Yup.

Reply:
For info, I've used this method of pre-setting the effects level in the mapping #2370 and #2061 as well to help with the lack of FX controls (pot or encoders) on these controllers. The idea is that I can then map a button or a pad to simply enable/disable the fx.

I had tried having the button command setting both the fx On/Off and level at the same times, but that often didn't work properly. That was early in the new FX Frameworks so maybe things have changed but I'm sure my limited coding knowledge might also have come into play.

Anyway, if there is a better method of doing things, I'm all for it, and removing it is not that much of an issue since this controller includes a Filter knob (where the two others mentioned didn't.)